editor.planx.uk/src/@planx/components/Filter/Editor.test.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  ComponentType as TYPES,
+  DEFAULT_FLAG_CATEGORY,
+  flatFlags,
+} from "@opensystemslab/planx-core/types";
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import React from "react";
+import { setup } from "testUtils";
+import { vi } from "vitest";
+
+import Filter from "./Editor";
+
+test("Adding a filter without explicit props uses the default flagset", async () => {
+  const handleSubmit = vi.fn();
+
+  setup(<Filter handleSubmit={handleSubmit} />);
+
+  expect(screen.getByTestId("flagset-category-select")).toHaveValue(
+    DEFAULT_FLAG_CATEGORY,
+  );
+
+  fireEvent.submit(screen.getByTestId("filter-component-form"));
+
+  await waitFor(() =>
+    expect(handleSubmit).toHaveBeenCalledWith(
+      {
+        type: TYPES.Filter,
+        data: {
+          fn: "flag",
+          category: DEFAULT_FLAG_CATEGORY,
+        },
+      },
+      mockDefaultFlagOptions,
+    ),
+  );
+});
+
+test("Adding a filter and selecting a flagset category", async () => {
+  const handleSubmit = vi.fn();
+
+  setup(<Filter handleSubmit={handleSubmit} />);
+
+  expect(screen.getByTestId("flagset-category-select")).toHaveValue(
+    DEFAULT_FLAG_CATEGORY,
+  );
+
+  fireEvent.change(screen.getByTestId("flagset-category-select"), {
+    target: { value: "Community infrastructure levy" },
```

Comment:
I'm using `fireEvent` throughout this file because the modal submit buttons like "Create filter" / "Update filter" aren't easily accessible to `user.click()` - therefore `fireEvent.submit()` is handy to have (this is commonly used throughout other "Editor" test files as well). 

I think the select interaction specifically could be migrated to `userEvent` going forward - which would mean we'd be testing the interaction more realistically to how the user would experience it in the browser.